pkg/remotecluster/remotecluster_test.go

```diff
@@ -0,0 +1,156 @@
+package remotecluster
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/klog"
+)
+
+const clusterID1 = "test-cluster-1"
+const pod0 = "pod0"
+
+var _ = Describe("Coastguard RemoteCluster", func() {
+	klog.InitFlags(nil)
+
+	var eventChannel chan Event
+
+	BeforeEach(func() {
+		eventChannel = make(chan Event, 10)
+	})
+
+	Describe("Synchronization", func() {
+
```

Comment:
I've found it a bit cleaner to create separate functions for each `Describe` block instead of inlining - makes the outer `Describe` function less monolithic (this is a bit of a downside to ginkgo IMO).

Reply:
I don't know, I saw your version, I like it more like this, or by making a function for each "it". But I'm ok with anything we decide, it's fine for me :)

Reply:
yeah or a function for each `It`. Whatever way is fine...

Reply:
Is it ok if I keep it as-is for now, I find it more readable and navigate (no need to jump to function), and also I don't need to pass the BeforeEach initialized variables. I understand this is very subjective.

Reply:
yeah it's fine either way.

Reply:
I favor the separation of each Describe instead of inlining as well

Reply:
:-P :-D

Reply:
It's a matter of style and preference. I think there's pros and cons to each - I don't have a strong preference either way so I'll let you guys debate it 😄

Reply:
I'm not a fan of separating in describes, but it's 2 against 1 ;) so I'll change it.

Reply:
I have refactored it a bit, I hope it will look better.